src/VolleyManagement.Backend/VolleyManagement.Domain/TournamentsAggregate/TournamentValidationSpecification.cs

```diff
@@ -14,37 +14,56 @@ public class TournamentValidationSpecification : ISpecification<Tournament>
         /// <returns> The <see cref="bool"/>. </returns>
         public bool IsSatisfiedBy(Tournament entity)
         {
-            // Name
-            if (string.IsNullOrEmpty(entity.Name) || entity.Name.Length >= Constants.Tournament.MAX_NAME_LENGTH)
+            if (IsSatisfiedByName(entity))
             {
                 return false;
             }
-            //// Description
-            if (!string.IsNullOrEmpty(entity.Description)
-                && entity.Description.Length >= Constants.Tournament.MAX_DESCRIPTION_LENGTH)
+            else if (IsSatisfiedByLocation(entity))
             {
                 return false;
             }
-            //// Location
-            if (!string.IsNullOrEmpty(entity.Location)
-                && entity.Location.Length >= Constants.Tournament.MAX_LOCATION_LENGTH)
+            else if (IsSatisfiedByDescription(entity))
             {
                 return false;
             }
-            //// Season
-            if (entity.Season <= Constants.Tournament.MINIMAL_SEASON_YEAR
-                && entity.Season >= Constants.Tournament.MAXIMAL_SEASON_YEAR)
+            else if (IsSatisfiedBySeason(entity))
             {
                 return false;
             }
-            //// Regulation link
-            if (!string.IsNullOrEmpty(entity.RegulationsLink)
-                  && entity.RegulationsLink.Length >= Constants.Tournament.MAX_REGULATION_LENGTH)
+            else if (IsSatisfiedByRegulationsLink(entity))
             {
                 return false;
             }
-
             return true;
         }
+
+        private static bool IsSatisfiedByName(Tournament entity)
+        {
+            return (string.IsNullOrEmpty(entity.Name) || entity.Name.Length >= Constants.Tournament.MAX_NAME_LENGTH);
+        }
+
+        private static bool IsSatisfiedByDescription(Tournament entity)
+        {           
+            return (!string.IsNullOrEmpty(entity.Description)
+                    && entity.Description.Length >= Constants.Tournament.MAX_DESCRIPTION_LENGTH);
```

Comment:
So if I pass null/empty string - it'll retun false. The same outcome awaits if I pass description with a valid length.

If you're checking if Description is invalid, try

`string.IsNullOrEmpty(entity.Description) || 
entity.Description.Length >= Constants.Tournament.MAX_DESCRIPTION_LENGTH);`